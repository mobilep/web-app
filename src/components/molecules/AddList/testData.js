export const users = [
	{
		id: 1,
		fullName: 'John Smith',
		initials: 'JS',
	}, {
		id: 2,
		fullName: 'Philippe Starck',
		initials: 'PS',
	}, {
		id: 3,
		fullName: 'Suzy Salmon',
		initials: 'SS',
	},
];

export const allUsers = [
	{
		id: 4,
		fullName: 'John Rebele',
		initials: 'JR',
	}, {
		id: 5,
		fullName: 'John Klokdan',
		initials: 'JK',
	}, {
		id: 6,
		fullName: 'John Makeba',
		initials: 'JM',
	},
];
