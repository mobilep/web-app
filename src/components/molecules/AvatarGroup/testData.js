export const avatars = [
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
];

export const avatarsLess = [
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
];
