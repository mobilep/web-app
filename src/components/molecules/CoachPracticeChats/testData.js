/* eslint-disable max-len */
export const practiceChats = [
	{
		currentUserId: '5d6e509093b6e912b83a65ce',
		_id: '5dad8ef5d3c420b10c618100',
		_coach: {
			_id: '5d6e509093b6e912b83a65ce',
			firstName: 'Roman',
			lastName: 'Yavoriv',
			avatarId: '5f549515-ee86-40f2-9104-5c6a8a80beb7',
			devices: 'Array[0]',
			avatarColor: '58c9ef',
			lang: 'en',
			avatar_sm: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/100/5f549515-ee86-40f2-9104-5c6a8a80beb7.jpg',
			avatar_md: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/640/5f549515-ee86-40f2-9104-5c6a8a80beb7.jpg',
			avatar_lg: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/1024/5f549515-ee86-40f2-9104-5c6a8a80beb7.jpg',
			havePassword: false,
			name: 'Roman Yavoriv',
			extraInformation: 'Array[7]',
		},
		_scenario: {
			_id: '5dad8ef546396073dd363907',
			groupChat: 'https://mobile-practice-dev.firebaseio.com/mpchat/5dad8ef546396073dd363909',
			isPracticed: true,
			isActive: true,
			name: '2nd scenario',
			video: 'Object',
		},
		_user: {
			_id: '5d70bc6d93b6e912b83a65d5',
			firstName: 'Taras',
			lastName: 'Kvas',
			avatarId: '2e3db607-9ec1-45f4-a7e5-4bed03912dd5',
			devices: 'Array[0]',
			avatarColor: 'fd524f',
			lang: 'en',
			avatar_sm: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/100/2e3db607-9ec1-45f4-a7e5-4bed03912dd5.jpg',
			avatar_md: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/640/2e3db607-9ec1-45f4-a7e5-4bed03912dd5.jpg',
			avatar_lg: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/1024/2e3db607-9ec1-45f4-a7e5-4bed03912dd5.jpg',
			havePassword: false,
			name: 'Taras Kvas',
			extraInformation: 'Array[7]',
		},
		__v: 0,
		createdAt: '2019-10-21T10:56:53.554Z',
		updatedAt: '2019-10-29T09:19:42.764Z',
		userEvaluatedAt: null,
		coachEvaluatedAt: null,
		hasBestPractice: false,
		hasUserMessage: true,
		isActive: true,
		lastMessageFrom: 'learner',
		status: 'current',
		coachMark: [],
		userMark: [],
		coachAudiosCount: 0,
		coachPhotosCount: 0,
		coachFilesCount: 0,
		coachVideosCount: 0,
		coachMessagesCount: 4,
		userAudiosCount: 0,
		userPhotosCount: 1,
		userFilesCount: 0,
		userVideosCount: 0,
		userMessagesCount: 19,
		_inbox: {
			_id: '5dad8ef546396073dd36390c',
			updatedAt: '2019-10-29T09:20:28.648Z',
			createdAt: '2019-10-21T10:56:53.544Z',
			_user: '5d6e509093b6e912b83a65ce',
			_recipient: '5d70bc6d93b6e912b83a65d5',
			__v: 0,
			inboxId: 'chats/5d6e509093b6e912b83a65ce/5dad8ef546396073dd363907/5dad8ef5d3c420b10c618100/5dad8ef546396073dd36390c',
			lastReadMessage: '-LsLo4OhHveQpLP5JXKy',
			unreadMessagesCount: 0,
			type: 'practice',
			status: 'read',
			message: 'Object',
			firstResponseAt: '2019-10-28T13:25:34.142Z',
			firstMessageAt: '2019-10-28T13:09:17.159Z',
			isActive: true,
		},
		state: 'waitingOnFeedback',
		messages: [
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
		],
	},
	{
		currentUserId: '5d6e509093b6e912b83a65ce',
		_id: '5dad8ef5d3c420b10c618102',
		_coach: {
			_id: '5d6e509093b6e912b83a65ce',
			firstName: 'Roman',
			lastName: 'Yavoriv',
			avatarId: '5f549515-ee86-40f2-9104-5c6a8a80beb7',
			devices: 'Array[0]',
			avatarColor: '58c9ef',
			lang: 'en',
			avatar_sm: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/100/5f549515-ee86-40f2-9104-5c6a8a80beb7.jpg',
			avatar_md: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/640/5f549515-ee86-40f2-9104-5c6a8a80beb7.jpg',
			avatar_lg: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/1024/5f549515-ee86-40f2-9104-5c6a8a80beb7.jpg',
			havePassword: false,
			name: 'Roman Yavoriv',
			extraInformation: 'Array[7]',
		},
		_scenario: {
			_id: '5dad8ef546396073dd363907',
			groupChat: 'https://mobile-practice-dev.firebaseio.com/mpchat/5dad8ef546396073dd363909',
			isPracticed: true,
			isActive: true,
			name: '2nd scenario',
			video: 'Object',
		},
		_user: {
			_id: '5da86a3716973b29347321ce',
			firstName: 'Roman',
			lastName: 'Yavoriv',
			devices: 'Array[0]',
			avatarColor: '00cdbb',
			lang: 'en',
			avatar_sm: null,
			avatar_md: null,
			avatar_lg: null,
			havePassword: false,
			name: 'Roman Yavoriv',
			extraInformation: 'Array[7]',
		},
		__v: 0,
		createdAt: '2019-10-21T10:56:53.563Z',
		updatedAt: '2019-10-29T09:18:36.340Z',
		userEvaluatedAt: '2019-10-29T09:16:14.949Z',
		coachEvaluatedAt: null,
		hasBestPractice: false,
		hasUserMessage: false,
		isActive: true,
		lastMessageFrom: 'coach',
		coachMark: [
			'Object',
		],
		userMark: [],
		coachAudiosCount: 0,
		coachPhotosCount: 0,
		coachFilesCount: 0,
		coachVideosCount: 0,
		coachMessagesCount: 2,
		userAudiosCount: 0,
		userPhotosCount: 0,
		userFilesCount: 0,
		userVideosCount: 0,
		userMessagesCount: 0,
		_inbox: {
			_id: '5dad8ef546396073dd36390d',
			updatedAt: '2019-10-29T09:19:11.848Z',
			createdAt: '2019-10-21T10:56:53.547Z',
			_user: '5d6e509093b6e912b83a65ce',
			_recipient: '5da86a3716973b29347321ce',
			__v: 0,
			inboxId: 'chats/5d6e509093b6e912b83a65ce/5dad8ef546396073dd363907/5dad8ef5d3c420b10c618102/5dad8ef546396073dd36390d',
			lastReadMessage: '-LsLnpAfH4s1IOZsQrgY',
			unreadMessagesCount: 0,
			type: 'practice',
			status: 'read',
			message: 'Object',
			firstResponseAt: null,
			firstMessageAt: '2019-10-29T09:18:27.588Z',
			isActive: true,
		},
		state: 'completed',
		messages: [
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
		],
	},
];

export const activeChatId = practiceChats[0]._id;
