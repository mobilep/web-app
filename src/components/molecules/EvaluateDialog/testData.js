export const evaluationList = [
	{
		id: '1',
		title: 'Contactâ€™s attention grabbed',
	}, {
		id: '2',
		title: 'Self Introduction',
	}, {
		id: '3',
		title: 'Purpose of call stated',
	},
];
