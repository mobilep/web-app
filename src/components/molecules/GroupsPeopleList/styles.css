.GroupPeopleList-empty {
	padding: 10px 20px;
	color: var(--grey3);
}
