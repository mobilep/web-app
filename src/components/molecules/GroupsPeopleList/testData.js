import { avatars } from '../AvatarGroup/testData';

export const groups = [
	{
		_id: '5b3b84595d7f59044c6759ea',
		name: 'Techmagic team',
		_users: avatars,
	},
	{
		_id: '5b3b84595d7f59044c6759eb',
		name: 'SoftServe team',
		_users: avatars,
	},
	{
		_id: '5b3b84595d7f59044c6759ec',
		name: 'Epam team',
		_users: avatars,
	},
];
