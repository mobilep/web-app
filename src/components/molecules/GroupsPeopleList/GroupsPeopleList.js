import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { GroupPeopleItem } from '..';

import './styles.css';

export default class GroupsPeopleList extends Component {

	static propTypes = {
		className: PropTypes.string,
		emptyGroupsText: PropTypes.string,
		groupList: PropTypes.array,
		onSelectGroup: PropTypes.func,
	}

	static defaultProps = {}

	generateList ({ groupList, onSelectGroup } = this.props) {
		return groupList.map((group) => {
			return (
				<GroupPeopleItem
					key={group._id}
					name={group._id}
					counter={group._users.length}
					people={group._users}
					header={group.name}
					selected={group.selected}
					onSelectGroup={() => onSelectGroup(group.selected, group._users)}
				/>
			);
		});
	}

	getClassName ({ className } = this.props) {
		return classNames('GroupsPeopleList', className);
	}

	render () {
		const className = this.getClassName();
		const { groupList, emptyGroupsText } = this.props;
		return (
			<div className={className}>
				{!groupList || groupList.length === 0
					? <div className='GroupPeopleList-empty'>{emptyGroupsText}</div>
					: this.generateList()}
			</div>
		);
	}
}
