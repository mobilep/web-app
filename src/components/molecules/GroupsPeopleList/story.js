import React from 'react';
import { storiesOf } from '@storybook/react';
import GroupsPeopleList from './GroupsPeopleList';
import { groups } from './testData';

storiesOf('2. Molecules / GroupsPeopleList', module)
	.add('default', () => <GroupsPeopleList groupList={groups} />);
