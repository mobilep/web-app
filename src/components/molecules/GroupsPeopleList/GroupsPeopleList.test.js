import React from 'react';
import { shallow } from 'enzyme';
import { groups } from '../GroupsPeopleList/testData';
import GroupsPeopleList from './GroupsPeopleList';

describe('Molecule - GroupsPeopleList', () => {
	it('Should render GroupsPeopleList component', () => {
		const wrapper = shallow(
			<GroupsPeopleList groupList={groups} />
		);
		expect(wrapper).toMatchSnapshot();
	});
});
