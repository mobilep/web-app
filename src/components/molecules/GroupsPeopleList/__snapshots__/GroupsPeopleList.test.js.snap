// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecule - GroupsPeopleList Should render GroupsPeopleList component 1`] = `
<div
  className="GroupsPeopleList"
>
  <GroupPeopleItem
    counter={6}
    header="Techmagic team"
    key="5b3b84595d7f59044c6759ea"
    name="5b3b84595d7f59044c6759ea"
    onSelectGroup={[Function]}
    people={
      Array [
        Object {
          "avatarColor": "#87d4ee",
          "firstName": "Clayton",
          "img": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B",
          "lastName": "Gross",
        },
        Object {
          "avatarColor": "#87d4ee",
          "firstName": "Clayton",
          "img": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B",
          "lastName": "Gross",
        },
        Object {
          "avatarColor": "#87d4ee",
          "firstName": "Clayton",
          "img": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B",
          "lastName": "Gross",
        },
        Object {
          "avatarColor": "#87d4ee",
          "firstName": "Clayton",
          "img": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B",
          "lastName": "Gross",
        },
        Object {
          "avatarColor": "#87d4ee",
          "firstName": "Clayton",
          "img": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B",
          "lastName": "Gross",
        },
        Object {
          "avatarColor": "#87d4ee",
          "firstName": "Clayton",
          "img": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B",
          "lastName": "Gross",
        },
      ]
    }
  />
  <GroupPeopleItem
    counter={6}
    header="SoftServe team"
    key="5b3b84595d7f59044c6759eb"
    name="5b3b84595d7f59044c6759eb"
    onSelectGroup={[Function]}
    people={
      Array [
        Object {
          "avatarColor": "#87d4ee",
          "firstName": "Clayton",
          "img": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B",
          "lastName": "Gross",
        },
        Object {
          "avatarColor": "#87d4ee",
          "firstName": "Clayton",
          "img": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B",
          "lastName": "Gross",
        },
        Object {
          "avatarColor": "#87d4ee",
          "firstName": "Clayton",
          "img": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B",
          "lastName": "Gross",
        },
        Object {
          "avatarColor": "#87d4ee",
          "firstName": "Clayton",
          "img": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B",
          "lastName": "Gross",
        },
        Object {
          "avatarColor": "#87d4ee",
          "firstName": "Clayton",
          "img": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B",
          "lastName": "Gross",
        },
        Object {
          "avatarColor": "#87d4ee",
          "firstName": "Clayton",
          "img": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B",
          "lastName": "Gross",
        },
      ]
    }
  />
  <GroupPeopleItem
    counter={6}
    header="Epam team"
    key="5b3b84595d7f59044c6759ec"
    name="5b3b84595d7f59044c6759ec"
    onSelectGroup={[Function]}
    people={
      Array [
        Object {
          "avatarColor": "#87d4ee",
          "firstName": "Clayton",
          "img": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B",
          "lastName": "Gross",
        },
        Object {
          "avatarColor": "#87d4ee",
          "firstName": "Clayton",
          "img": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B",
          "lastName": "Gross",
        },
        Object {
          "avatarColor": "#87d4ee",
          "firstName": "Clayton",
          "img": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B",
          "lastName": "Gross",
        },
        Object {
          "avatarColor": "#87d4ee",
          "firstName": "Clayton",
          "img": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B",
          "lastName": "Gross",
        },
        Object {
          "avatarColor": "#87d4ee",
          "firstName": "Clayton",
          "img": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B",
          "lastName": "Gross",
        },
        Object {
          "avatarColor": "#87d4ee",
          "firstName": "Clayton",
          "img": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B",
          "lastName": "Gross",
        },
      ]
    }
  />
</div>
`;
